<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
 <HEAD>
   <TITLE> [antlr-interest] Wondering about: Error handling
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:antlr-interest%40antlr.org?Subject=Re:%20%5Bantlr-interest%5D%20Wondering%20about%3A%20Error%20handling&In-Reply-To=%3C9da4f4520706171313u2d9e6d19kf1b6a4dc836c3518%40mail.gmail.com%3E">
   <META NAME="robots" CONTENT="index,nofollow">
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="021453.html">
   <LINK REL="Next"  HREF="021456.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[antlr-interest] Wondering about: Error handling</H1>
    <B>Oliver Zeigermann</B> 
    <A HREF="mailto:antlr-interest%40antlr.org?Subject=Re:%20%5Bantlr-interest%5D%20Wondering%20about%3A%20Error%20handling&In-Reply-To=%3C9da4f4520706171313u2d9e6d19kf1b6a4dc836c3518%40mail.gmail.com%3E"
       TITLE="[antlr-interest] Wondering about: Error handling">oliver at zeigermann.de
       </A><BR>
    <I>Sun Jun 17 13:13:39 PDT 2007</I>
    <P><UL>
        <LI>Previous message: <A HREF="021453.html">[antlr-interest] Error with date
</A></li>
        <LI>Next message: <A HREF="021456.html">[antlr-interest] Wondering about: Error handling
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#21448">[ date ]</a>
              <a href="thread.html#21448">[ thread ]</a>
              <a href="subject.html#21448">[ subject ]</a>
              <a href="author.html#21448">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>Folks!

I understand that - by default - ANTLR generates something like this

        catch (RecognitionException re) {
            reportError(re);
            recover(input,re);
        }


as generic code for error handling. When you want something else you
can specify your own error handling block using

catch [...] {
...
}

When I want special handling for certain exceptions only I can specify
that in the catch phrase. Like that:

catch [FailedPredicateException fpe] {
...
}

However, this replaces the orignal catch block instead of being added.
I expected something like this to be generated:


     catch (FailedPredicateException fpe) {
       ...
        }
      catch (RecognitionException re) {
            reportError(re);
            recover(input,re);
        }

But only the first block actually is generated.

This leads to problems when &quot;normal&quot; recognition exceptions occur as
they are no longer handled by the default procedure, but rather are
passed on to calling rules.

Is this intented behavior? Does it make sense to have it like that?

Just wondering and cheers

Oliver
</PRE>
















<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="021453.html">[antlr-interest] Error with date
</A></li>
	<LI>Next message: <A HREF="021456.html">[antlr-interest] Wondering about: Error handling
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#21448">[ date ]</a>
              <a href="thread.html#21448">[ thread ]</a>
              <a href="subject.html#21448">[ subject ]</a>
              <a href="author.html#21448">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="http://www.antlr.org/mailman/listinfo/antlr-interest">More information about the antlr-interest
mailing list</a><br>
</body></html>
